src/common/components/search/SearchBar.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { FC, useState, useRef, useEffect } from 'react';
+import { Search } from '../icons/Search';
+import LoadingSearchItem from './LoadingSearchItem';
+import { useBasicSearchLazyQuery } from 'src/graphql/indexerTypes';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import * as zod from 'zod';
+import SearchResults from './SearchResults';
+import { useOutsideAlerter } from '@/common/hooks/useOutsideAlerter';
+import { useRouter } from 'next/router';
+import { Close } from '../icons/Close';
+
+const schema = zod.object({
+    query: zod.string().nonempty({message: `Must enter something`})
+})
+
+interface SearchQuerySchema {
+    query: string;
+}
+
+const SearchBar: FC = () => {
+
+    const searchResultsRef = useRef<HTMLDivElement>(null!);
+
+    const router = useRouter()
+    
+
+    const [showBar, setShowBar] = useState(false)
+    const [showResults, setShowResults] = useState(false)
+
+    useEffect(() => {
+        setShowResults(false)
+    }, [router.route])
+
+    useOutsideAlerter(searchResultsRef, () => setShowResults(false))
+
+    const toggleBar = () => {
+        setShowBar(!showBar)
+    }
+
+    const [searchQuery, { data, loading, called, refetch }] = useBasicSearchLazyQuery()
+
+    const handleSearch = ({query}: SearchQuerySchema) => {
+        setShowResults(true)
+        searchQuery({variables: {
+            walletAddress: query
+        }})
+    }
+
+    const handleReset = () => {
+        setValue("query", "")
+        setShowResults(false)
+    }
+
+    const { 
+        handleSubmit, 
+        setValue,
+        register,
+        watch,
+        formState: {isSubmitting}
+    } = useForm<SearchQuerySchema>({
+        resolver: zodResolver(schema)
+    })
+
+    const searchQueryWatchable = watch("query")
+
+
+    return (
+        <div id={`searchbar-container`} ref={searchResultsRef} className={`flex flex-row items-center w-full relative z-30`}>
+            {!showBar ? (
+                <a onClick={toggleBar} className={`hover:bg-gray-800 p-2 ease-in-out transition rounded-full hover:cursor-pointer`}>
+                    <Search/>
+                </a>
+            ) : (
+                <div className={`flex flex-row gap-2 items-center p-2 rounded-full bg-gray-900 border border-white w-full`}>
+                    <a onClick={toggleBar} className={``}>
+                        <Search/>
+                    </a>
+                    <form className={`flex items-center w-full relative`} onSubmit={handleSubmit(handleSearch)}>
+                        <input
+                        {...register('query', {required: true})}
```

Comment:
should this block be indented under `<input`?

Reply:
prettier